src/index.ts

```diff
@@ -0,0 +1,537 @@
+   * It defaults to "bump", which means the commit will be treated as a version bump,
+   * if it contains one of the conventional commit types.
+   *
+   * "ignore" will not treat this commit as a version bump for a package
+   * that is not in the scope.
+   * @default "bump"
+   */
+  nonScopeBehavior?: NonScopeBehavior;
+}
+
+interface Package {
+  name: string;
+  tagPrefix: string;
+  directory: string;
+  dependsOn: string[];
+}
+
+interface VersionChanges {
+  major: boolean;
+  minor: boolean;
+  patch: boolean;
+}
+
+interface VersionUpdate {
+  tagPrefix: string;
+  currentVersion: string;
+  nextVersion: string;
+  hasChanges: boolean;
+  changes: CommitInfo[];
+}
+
+interface CommitInfo {
+  hash: string;
+  message: string;
+  type: string;
+  breaking: boolean;
+  reasons: string[];
+}
+
+// Load configuration
+const CONFIG_PATH = resolve(process.cwd(), "release-config.json");
+let config: Config | undefined;
+
+// Add color support
+const colors = {
+  reset: "\x1b[0m",
+  bright: "\x1b[1m",
+  dim: "\x1b[2m",
+  red: "\x1b[31m",
+  green: "\x1b[32m",
+  yellow: "\x1b[33m",
+  blue: "\x1b[34m",
+  magenta: "\x1b[35m",
+  cyan: "\x1b[36m",
+  white: "\x1b[37m",
+} as const;
+
+function formatCommit(hash: string, message: string): string {
+  return `${colors.yellow}${hash.slice(0, 7)}${colors.reset} ${message}`;
+}
+
+function printSection(title: string, content: string = ""): void {
+  console.log(`\n${colors.bright}${colors.blue}${title}${colors.reset}`);
+  if (content) {
+    console.log(content);
+  }
+}
+
+function getLastTag(prefix: string): string {
+  try {
+    // For root packages with just 'v', we need to be more careful about the match pattern
+    const matchPattern = prefix === "v" ? "v[0-9]*" : `${prefix}*`;
+    // Redirect stderr to /dev/null to suppress "fatal: No names found" message
+    return execSync(
+      `git describe --tags --match "${matchPattern}" --abbrev=0 2>/dev/null`,
+    )
+      .toString()
+      .trim();
+  } catch {
+    return "";
+  }
+}
+
+function getCommitRange(prefix: string): string {
+  const lastTag = getLastTag(prefix);
+  return lastTag ? `${lastTag}..HEAD` : "HEAD";
+}
+
+function getCurrentVersion(prefix: string): string {
+  const lastTag = getLastTag(prefix);
+  if (!lastTag) return "0.0.0";
+  return lastTag.replace(prefix, "");
+}
+
+function determineVersionChange(commits: string[]): VersionChanges {
+  const changes: VersionChanges = {
+    major: false,
+    minor: false,
+    patch: false,
+  };
+
+  for (const commit of commits) {
+    const message = commit.toLowerCase();
+    if (
+      message.startsWith("breaking") ||
+      message.includes("breaking change") ||
+      message.includes("breaking-change")
+    ) {
+      changes.major = true;
+    } else if (message.startsWith("feat")) {
+      changes.minor = true;
+    } else if (
+      message.startsWith("fix") ||
+      message.startsWith("docs") ||
```

Comment:
This is definitely NOT correct. Only fix cause a patch bump, not any of the others.